libtiledbsoma/src/utils/column_buffer.cc

```diff
@@ -42,26 +42,31 @@ using namespace tiledb;
 
 std::shared_ptr<ColumnBuffer> ColumnBuffer::create(
     std::shared_ptr<Array> array, std::string_view name) {
-    return ColumnBuffer::create(array->schema(), name);
-}
-
-std::shared_ptr<ColumnBuffer> ColumnBuffer::create(
-    ArraySchema schema, std::string_view name) {
+    auto schema = array->schema();
     auto name_str = std::string(name);  // string for TileDB API
 
     if (schema.has_attribute(name_str)) {
```

Comment:
I so far have also limited myself to attributes but I think the core implementation is more powerful and we could have enumerations as dimensions (!!) given that we already have both int and char dims anyway.

Reply:
let's double-check with @davisp -- ?

Reply:
Big nopes to enumeration support for dimensions. The primitives are close, but there are a whole bunch of non-obvious behavior issues we’d run into due to dimensions requiring a defined sort and a bunch of other things I don’t know enough about to even begin listing.

Which is to say that its probably not technically infeasible, but the consensus at the design stage was call dimension support out of scope for the MVP/V1 implementation.

Reply:
Thank you so much for clarifying.  I have been running so far with 'attributes only' too but must have hallucinated myself into thinking you had mused about dims too.  I probably poorly inferred from query conditions.   All good then.

Reply:
No worries! This is absolutely one of those situations where, given the primitives, it seems like adding the functionality would be straightforward, but then when we start working through the edge cases there are a whole bunch of open questions on what the behavior should be with no clear answers.

The current implementation ignores dimensions purely out of expediency to get something usable first which both makes sure that its usable as designed and should also help shake out anything that was overlooked. I’d wager a small amount of money that enumerations for dimensions will exist eventually, its just a matter of making sure our behaviors are all intentional for long term support.

Reply:
Man you make me look good because I did use the lawyer-advised (kidding here) conditional above:

> and we could have enumerations as dimensions